src/test/java/net/kaczmarzyk/spring/data/jpa/utils/PathVariableResolverTest.java

```diff
@@ -19,46 +19,56 @@
 
 import org.junit.Test;
 
+import java.util.Map;
+
 public class PathVariableResolverTest {
 
 	@Test
 	public void extractsPathVariable() {
-		PathVariableResolver resolver = PathVariableResolver
-				.forPathPatternAndActualPath("/customers/{customerId}", "/customers/77");
+		Map<String, String> resolvedVariables = PathVariableResolver
+				.resolvePathVariables("/customers/{customerId}", "/customers/77");
+
+		assertThat(resolvedVariables.get("customerId")).isEqualTo("77");
+	}
+
+	@Test
+	public void extractPathVariableWithRegex() {
+		Map<String, String> resolvedVariables = PathVariableResolver
+				.resolvePathVariables("/customers/{customerId:[0-9]+}", "/customers/77");
 
-		assertThat(resolver.resolveValue("customerId")).isEqualTo("77");
+		assertThat(resolvedVariables.get("customerId")).isEqualTo("77");
 	}
 
 	@Test
 	public void extractsPathVariable_untilSlash() {
-		PathVariableResolver resolver = PathVariableResolver
-				.forPathPatternAndActualPath("/customers/{customerId}/orders", "/customers/77/orders");
+		Map<String, String> resolvedVariables = PathVariableResolver
+				.resolvePathVariables("/customers/{customerId}/orders", "/customers/77/orders");
 
-		assertThat(resolver.resolveValue("customerId")).isEqualTo("77");
+		assertThat(resolvedVariables.get("customerId")).isEqualTo("77");
 	}
 
 	@Test
 	public void extractsPathVariableAmongMultiplePathVariables() {
```

Comment:
I think we need to explicitly test scenario when url does not match the pattern e.g.

/customers/{id:[0-9]}
vs
/customers/notANumber